analyzers/src/SonarAnalyzer.CSharp/Rules/ClassAndMethodName.cs

```diff
@@ -128,10 +130,10 @@ protected override void Initialize(SonarAnalysisContext context)
                 SyntaxKind.ClassDeclaration,
                 SyntaxKind.InterfaceDeclaration,
                 SyntaxKind.StructDeclaration,
-                SyntaxKindEx.RecordClassDeclaration);
+                SyntaxKindEx.RecordClassDeclaration,
+                SyntaxKindEx.RecordStructDeclaration);
 
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
```

Comment:
There is a `ClassDeclarationSyntax` check in line 166:
`if (typeDeclaration is ClassDeclarationSyntax && IsTestClassName(typeDeclaration.Identifier.ValueText))`

https://github.com/SonarSource/sonar-dotnet/blob/a10e7dc4555267a726ff334ff654481a472fb5ab/analyzers/src/SonarAnalyzer.CSharp/Rules/ClassAndMethodName.cs#L166-L169

Are there any test frameworks with support for `record` and/or `record struct` as test method containers (It just works probably)? Is anybody doing this?

Reply:
I don't know. My guess is it would work just fine for `record`. And at the same time I don't see a reason why would somebody use that.

I would leave it as it is now. We can change it if we see an issue raised by users.